### PR TITLE
Add 'Throws' tests for Named and Anonymous Pipes to increase coverage.

### LINKED ISSUE #1142: Improve code coverage for System.IO.Pipes
According to the current code coverage report, this library is only at ~25%. Our general goal is closer to 90%, though this will depend on the library. We should investigate to see whether the current coverage level is appropriate, and if not, improve our test suites to provide better coverage.

### PR DESCRIPTION
**Pull request by bpschoch**
This is first set of tests concentrating on exception throwing

partial Fix #1142

### CODE REVIEW

File: src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+{
+    public class AnonymousPipesThrowsTests
+    {
+        // Server parameter validation tests
+        [Fact]
+        public static void ServerBadPipeDirectionThrows()
+        {
+            Assert.Throws<NotSupportedException>(delegate
+            {
+                using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.InOut, System.IO.HandleInheritability.None))
```
Comment (Maxwe11) [inline src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs]: there is no need in `using`

Comment (bpschoch) [inline src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs]: True, I guess good code habits are a bit too ingrained (ie about using a using).  I will work on taking those out.

File: src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+                Assert.Throws<ArgumentNullException>(() => server.Write(null, -1, 1));
+            }
+        }
+
+        [Fact]
+        public async static void ServerWriteAsyncNegativeOffsetThrows()
+        {
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                await Assert.ThrowsAsync<ArgumentOutOfRangeException>(async () => await server.WriteAsync(new byte[5], -1, 1));
```
Comment (stephentoub) [inline src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs]: The async on the lambda and the await in the delegate aren't actually necessary here... the delegate just needs to return a Task, which WriteAsync does.

File: src/System.IO.Pipes/tests/System.IO.Pipes.Tests.csproj [hunk 3]
```diff
@@ -20,9 +20,11 @@
   </PropertyGroup>
   <!-- Compiled Source Files -->
   <ItemGroup>
+    <Compile Include="AnonymousPipesThrowsTests.cs" />
     <Compile Include="NamedPipesSimpleTest.cs" />
     <Compile Include="AnonymousPipesSimpleTest.cs" />
     <Compile Include="$(CommonPath)\Interop\Interop.PlatformDetection.cs" />
+    <Compile Include="NamedPipesThrowsTests.cs" />
```
Comment (stephentoub) [inline src/System.IO.Pipes/tests/System.IO.Pipes.Tests.csproj]: I realize it already wasn't sorted as such, but could you alphabetize the list?

**Comment by stephentoub** [comment]
Thanks for adding these.  Left a few comments.  Otherwise LGTM.

**Comment by bpschoch** [comment]
I managed to mess up on git which caused the merge conflict (I'm still learning), I think I fixed it, and pushed force which is what I should have done in the first place.  I plan on working on more tests so I'll fix the sort order then.

**Comment by Maxwe11** [comment]
failed due to #1434

**Comment by bpschoch** [comment]
I"m not sure what to do, I only added 2 new test modules to system.io.pipes and a test fails in system.io.filesystem which is unrelated to my changes?

**Comment by Maxwe11** [comment]
>I"m not sure what to do

Wait until someone fix that issue :smile:

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by bpschoch** [comment]
The only thing that I can see is common with system.io.pipes and system.io.filesystem are the use of native safehandles;  Is there an issue of running out of safehandles?

**Comment by stephentoub** [comment]
@bpschoch, it's not due to your change; it's failing in the main build as well.

**Comment by stephentoub** [comment]
@bpschoch, can you address the async/awaits comment? We can then merge it even with the CI failure as long as it's this same one not caused by your change.

**Comment by bpschoch** [comment]
took out the async and await in the Assert.ThrowsAsync (I think this is what was asked) and re-ordered files in project

**Comment by ellismg** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Thanks, @bpschoch. LGTM.

**Pull request merged**